apps/dashboard/src/components/primitives/field-editor/variable-popover/hooks/use-transformer-manager.ts

```diff
@@ -0,0 +1,113 @@
+import { useCallback, useState } from 'react';
+import { TRANSFORMERS } from '../constants';
+import type { TransformerWithParam } from '../types';
+
+type UseTransformerManagerProps = {
+  initialTransformers: TransformerWithParam[];
+  onUpdate: (transformers: TransformerWithParam[]) => void;
+};
+
+export function useTransformerManager({ initialTransformers, onUpdate }: UseTransformerManagerProps) {
```

Comment:
It feels that these hooks belong to the sortable component and they need to be generalized to control the sortable state.